src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -0,0 +1,147 @@
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string output = GenerateCompiledParametersFileAndReturnOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(output);
+        }
+
+        private string GenerateCompiledParametersFileAndReturnOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            string compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepFilePath);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the generate params.
+            if (File.Exists(compiledFilePath) && !TemplateContainsBicepGeneratorMetadata(File.ReadAllText(compiledFilePath)))
+            {
+                return "Generate Params failed. The file \"" + compiledFile + "\" already exists and was not generated by Bicep. If overwriting the file is intended, delete it manually and retry the Generate Params command.";
+            }
+
+            var fileUri = documentUri.ToUri();
+            RootConfiguration? configuration = null;
+
+            try
+            {
+                configuration = this.configurationManager.GetConfiguration(fileUri);
+            }
+            catch (ConfigurationException exception)
+            {
+                // Fail the generate params if there's configuration errors.
+                return exception.Message;
+            }
+
+            CompilationContext? context = compilationManager.GetCompilation(fileUri);
+            Compilation compilation;
+
+            if (context is null)
+            {
+                SourceFileGrouping sourceFileGrouping = SourceFileGroupingBuilder.Build(this.fileResolver, this.moduleDispatcher, new Workspace(), fileUri, configuration);
+                compilation = new Compilation(features, namespaceProvider, sourceFileGrouping, configuration, new LinterAnalyzer(configuration));
+            }
+            else
+            {
+                compilation = context.Compilation;
+            }
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                return "Generate Params failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            using var fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite);
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), emitterSettings);
+            EmitResult result = emitter.EmitParametersFile(fileStream);
+
+            return "Generate Params succeeded. Created file " + compiledFile;
```

Comment:
If the file already exists, it should say "Updated file"...

Reply:
changed the message to `processed file`

Reply:
I guess good enough for now.